pkg/logmgr/logmgr.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logmgr
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/vmware/vic/lib/tether"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// RotateInterval defines a type for a log rotate frequency.
+type RotateInterval uint32
+
+// LogRotateBinary points to a logrotate path in the system.
+const LogRotateBinary = "/usr/sbin/logrotate"
+
+const (
+	// Daily to trim logs daily.
+	Daily RotateInterval = iota
+	// Hourly to trim logs hourly.
+	Hourly
+	// Weekly to trim logs weekly.
+	Weekly
+	// Monthly to trim logs monthly.
+	Monthly
+)
+
+type logRotateConfig struct {
+	rotateInterval RotateInterval
+	logFilePath    string
+	logFileName    string
+	maxLogSize     int64
+	maxLogFiles    int64
+	compress       bool
+}
+
+// ConfigFileContent formats log configuration according to logrotate requirements.
+func (lrc *logRotateConfig) ConfigFileContent() string {
```

Comment:
You could just use json for the config file format and let Go do most of the marshaling/unmarshaling.